text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,233 @@
+
+```html
+<!-- x/foo.html -->
+<template>
+  <div>Hello</div>
+</template>
+```
+
+```js
+// x/foo.js
+import { LightningElement } from 'lwc'
+
+export default class Foo extends LightningElement {
+  static shadow = false
+}
+```
+
+```css
+/* x/foo.scoped.css */
+div {
+  color: green;
+}
+```
+
+The result will be:
+
+```html
+<x-app>
+  <style class="x-foo_foo" type="text/css">
+    div.x-foo_foo { color: green; }
+  </style>
+  <div class="x-foo_foo">Hello</div>
+</x-app>
+```
+
+## Motivation
+
+For Light DOM ([#44](https://github.com/salesforce/lwc-rfcs/pull/44)), the default assumption is that styles are unscoped. In other words, a light DOM component essentially just contains `<style>` tags that are inserted into the DOM _in situ_. (The actual implementation may be slightly different, but this is the basic mental model.)
+
+Many frameworks, however, have a concept of _scoped styles_, even without using shadow DOM. These offer good developer ergonomics, because developers can concentrate on the CSS co-located with a particular component, without worrying how that CSS might affect other components.
+
+Prior art:
+
+- [Vue: `<style scoped>`](https://vue-loader.vuejs.org/guide/scoped-css.html)
+- [Svelte: scoped styles](https://svelte.dev/docs#style)
+- [React: CSS Modules in `create-react-app`](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/)
+- [Stencil: scoped styles in light DOM](https://stenciljs.com/docs/styling#scoped-css)
+- [BEM: Block, Element, Modifier](https://en.bem.info/)
+
+## Detailed design
+
+### Scoping token
+
+For the purposes of this document, a _scoping token_ is some string that we use to scope CSS selectors to a particular region of the DOM.
+
+In general, there are two approaches for applying scoping tokens to the DOM: classes or attributes. Both have [the same CSS specificity](https://alistapart.com/article/braces-to-pixels/#section4), but [historically](https://github.com/threepointone/glamor/issues/339) and [presently](https://github.com/salesforce/lwc/pull/2329), classes are faster than attributes in terms of the browser's [style calculation](https://developers.google.com/web/fundamentals/performance/rendering/#the_pixel_pipeline) process. So we prefer classes.
+
+This means that, with scoped styles, all CSS selectors inside of the `*.scoped.css` file will have an added class, which scopes the rules to that particular component.
+
+For example, the user might author:
+
+```css
+div > .bar {}
+```
+
+This is compiled to:
+
+```css
+div.x-foo_foo > .bar.x-foo_foo {}
+```
+
+Any existing classes in the template will be merged with the scoping token classes.
+
+### Comparisons with synthetic shadow DOM
+
+The current design of LWC's synthetic shadow scoped styles has several features that we do not want to emulate with light DOM scoped styles:
+
+- Using attributes over classes.
+- Allowing `@import` to import other CSS files.
+- Evaluating scoped CSS at runtime rather than compile time.

```

Comment:
I mean the fact that we generate a function that takes the scoping token as input and outputs a string, rather than just generating a string server-side. I'll make it clearer here.